frontend/app/helpers/useLoginValidation.ts

```diff
@@ -0,0 +1,73 @@
+import { useState } from "react"
+
+interface UserCredentials {
+  username: string
+  password: string
+}
+
+interface ValidationError {
+  username: string
+  password: string
+}
+
```

Comment:
Mentioned in the previous comments, please keep types at the end of the file

Reply:
sorry i had done it but when i moved the code to create a custome hook i forgot